ament_tools/build_types/cmake.py

```diff
@@ -149,20 +173,24 @@ def _common_cmake_on_build(self, should_run_configure, context, prefix, extra_cm
                 if vsv not in supported_vsv:
                     raise VerbExecutionError('Unknown / unsupported VS version: ' + vsv)
                 cmake_args += ['-G', supported_vsv[vsv]]
+            elif IS_MACOSX:
+                if context.use_xcode or self._using_xcode_generator(context):
+                    cmake_args += ['-G', 'Xcode']
+                    cmake_args += ['-DCMAKE_XCODE_GENERATE_SCHEME=ON']
```

Comment:
This one assumes CMake 3.8.0 right? Could we conditionally add that with that version only? Or is there no point in running without it? My impression was that it just improved testing with Xcode?